### ISSUE TITLE
Accessibility Bug: When the message strips are accessed, the information inside the message strips are not conveyed (screen reader)

### ISSUE THREAD

**Issue by puru-hk** [SAP/fundamental-ngx#3123]
Fundamental core component: "Message Strip"

Description: When the message strips are accessed, the information inside the message strips are not conveyed (screen reader)

Expected: The content inside the message strips should be conveyed properly when the message strips are accessed(screen reader)

Attached the screenshot

![image](https://user-images.githubusercontent.com/32538291/91556741-db0ded80-e950-11ea-8a1e-17d2464ab595.png)

Note: - Issue persists for all the message strips present within the ‘Message strip’ control.

**Comment by puru-hk** [comment]
@Sanjay-1312 Please test this issue in this url https://deploy-preview-5458--fundamental-ngx.netlify.app

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

The issue is not fixed, hence i am re opening the issue.
On accessing the 'Message Strip', the close information is only read, the message strip information is not read.
There is no keyboard support for the message strip where there is Close button is not available.

URL: https://deploy-preview-5458--fundamental-ngx.netlify.app
JAWS 2021

Screenshot :
![image](https://user-images.githubusercontent.com/78714518/120278503-fee0d400-c2d2-11eb-998a-5343a9b34d79.png)

**Comment by puru-hk** [comment]
@kavya-b This issue is not fixed it-seems. Kindly have a look.

**Comment by kavya-b** [comment]
As discussed with @puru-hk and @Sanjay-1312, the only way we see to fix this in JAWS is to provide `tabindex="0"` on the message strip. This means adding focus on message strip, which is not specified in the Fiori3 visual specs. Without this, even the UI5 example for message strip does not read the strip if a close button is not present.
Another issue is that UI5 reads the entire message strip content on tabbing to the close button. Is this correct behaviour? Shouldn't a button read only what it is trying to indicate rather than the entire strip's contents? Also the message strip should read what it is trying to indicate. I think this is a question for designers as well. cc: @manjunathanagaraj @kenykim 

@Sanjay-1312 will follow up with his team to understand whether we need focus on the entire message strip and how can they test a strip without a close button, if it is expected that close button will read out the entire strip's information.

**Comment by kavya-b** [comment]
@InnaAtanasova @droshev currently, we do not have focus specification by Fiori for Message Strip. However, it seems like this is necessary in order for screen readers(especially JAWS) to read the message strip content. 

Can you let us know if we can get the focus specs added to Message Strip visual specs or if we can use the general Focus Visual Concept spec to apply focus to Message Strip in order to address this a11y issue? It would also help if we get feedback from design team regarding this. Please help.

This PR and issue is blocked until we are able to add focus to Message Strip container.
cc: @manjunathanagaraj

**Comment by droshev** [comment]
@kavya-b the content in the message strip is custom and it is up to application developers to create. If the use case is to have focusable elements then the reader app will focus and read. We don't control what it is inside.

**Comment by kavya-b** [comment]
@droshev I agree we don't have control on the content, but the a11y team's requirements say the following:
- when a close button is present for the strip, tabbing to the button should read the entire strip content, the status of the strip and the button information.
- when a close button is not present for the strip, they verify only the static elements(message strip text etc) separately through a different mode of JAWS navigation

this way, focus on the whole message strip is no longer required, but extra information would have to be read out from the close button.

request @Sanjay-1312 to add your thoughts and suggest if @droshev 's comment is okay then we can keep this issue closed, or reopen this issue as needed.

cc: @puru-hk @manjunathanagaraj

**Comment by Veerakarthick** [comment]
Hello Colleague ,

Retest has been taken reported issue still persist the same.
Description: When the message strips are accessed, the information inside the message strips are not conveyed (screen reader)

Expected: The content inside the message strips should be conveyed properly when the message strips are accessed(screen reader)
Screenshot:
![image](https://user-images.githubusercontent.com/37998745/151794695-1138cf1d-f894-460f-9d97-fc7f9294916f.png)

REgards,
Karthikeyan.c

**Comment by Veerakarthick** [comment]
Hello Colleague,

Retest has been taken reported issue still persist the same so please check and do the needful.
Description: When the message strips are accessed, the information inside the message strips are not conveyed (screen reader)

Expected: The content inside the message strips should be conveyed properly when the message strips are accessed(screen reader)

Screenshot:
![image](https://user-images.githubusercontent.com/37998745/159398889-f5db029b-dce8-430f-8188-596f461fb615.png)

Retest envornment:
URL: https://deploy-preview-5458--fundamental-ngx.netlify.app/
JAWS 2021

Regards,
Karthikeyan.c

**Comment by Karthickc1994** [comment]
Hello Colleague,

Retest has been taken with the below environment test results are below:

Issue: When the message strip is accessed the information within the message strip is not read out.
Reading made as:
Close  Button  
To activate press spacebar.
Close  Button  
To activate press spacebar.
Close  Button  
To activate press spacebar.
 Show Code  Button  
To activate press spacebar.

Screen shot:
![image](https://user-images.githubusercontent.com/105037808/167371334-3f597049-ae93-49a6-8789-48dcefb2e43c.png)

Expected behavior: 
when the message strip is accessed the information within the message strip should read out by the screen reader.

Retest Environment:
URL:[https://deploy-preview-5458--fundamental-ngx.netlify.app](https://deploy-preview-5458--fundamental-ngx.netlify.app/?utm_source=github&utm_campaign=bot_dp)
Browser: Version 101.0.4951.54 (Official Build) (64-bit)
Screen Reader: JAWS 2022.2204.20 ILM

Regards,
Karthikeyan.c
